Core/Core/Inbox/ComposeMessage/ViewModel/ComposeMessageViewModel.swift

```diff
@@ -67,38 +67,98 @@ class ComposeMessageViewModel: ObservableObject {
     @Published public var includedMessages: [ConversationMessage] = []
     @Published public var attachments: [File] = []
     @Published public var isShowingCancelDialog = false
+    @Published var showSearchRecipientsView: Bool = false
     public let confirmAlert = ConfirmationAlertViewModel(
         title: String(localized: "Unsaved Changes", bundle: .core),
         message: String(localized: "You have unsaved changes in your message. If you leave now, your current message will be lost.", bundle: .core),
         cancelButtonTitle: String(localized: "Cancel", bundle: .core),
         confirmButtonTitle: String(localized: "Discard", bundle: .core),
         isDestructive: true
     )
+
+     let errorAlert = ConfirmationAlertViewModel(
+        title: String(localized: "Oops!", bundle: .core),
+        message: String(localized: "This message could not be sent. Tap to try again.", bundle: .core),
+        cancelButtonTitle: String(localized: "Back to editing", bundle: .core),
+        confirmButtonTitle: String(localized: "Retry", bundle: .core),
+        isDestructive: false
+    )
     let router: Router
 
     // MARK: - Private
+    private var didSentMailSuccessfully: PassthroughSubject<Void, Never>?
+    public let didTapRetry = PassthroughRelay<WeakViewController>()
+    private var viewController  = WeakViewController()
     private var subscriptions = Set<AnyCancellable>()
     private let interactor: ComposeMessageInteractor
+    private let recipientUseCase: RecipientInteractor
+    private let audioSession: AudioSessionProtocol
+    private let cameraPermission: CameraPermissionService.Type
     private let scheduler: AnySchedulerOf<DispatchQueue>
     private var messageType: ComposeMessageOptions.MessageType
-
+    private var allRecipients = CurrentValueSubject<[Recipient], Never>([])
     private var hiddenMessage: String = ""
     private var autoTeacherSelect: Bool = false
     private var teacherOnly: Bool = false
 
     // MARK: Public interface
-
-    public init(router: Router, options: ComposeMessageOptions, interactor: ComposeMessageInteractor, scheduler: AnySchedulerOf<DispatchQueue> = .main) {
+    public init(router: Router,
+                options: ComposeMessageOptions,
+                interactor: ComposeMessageInteractor,
+                scheduler: AnySchedulerOf<DispatchQueue> = .main,
+                recipientUseCase: RecipientInteractor,
```

Comment:
```suggestion
                recipientInteractor: RecipientInteractor,
```
Please update other occurrences accordingly.